### PR TITLE
Tidy up a bit.

### PR DESCRIPTION
**Pull request by vext01**
Misc tidying and a couple of bug fixes. Most of this found by static analyser.

On a related note: I've been using this particular analyser (flake8 + vim plugin) for a year or so on my own projects. It reports many other cosmetic changes (long lines, whitespace, etc.) that we could apply, however I have restrained since we don't really have a coding style in place.

Since we are now in the business of improving how we do software engineering, Is there any interest in adopting a coding style? PEP8 is the standard for Python code, and most static analysers and IDEs support it out the box. I would be willing to give it a shot.

If there is interest in this, we should not, however, enforce this on external projects.

Cheers

### CODE REVIEW

**Comment by ltratt** [comment]
I am all for using widely adopted standards (provided they're not completely obnoxious).

File: krun.py [hunk 1]
```diff
@@ -232,7 +229,7 @@ def run(self):
 
         info("Done: Results dumped to %s" % self.out_file)
         if errors:
-            warn("%s ERRORS OCCURRED! READ THE LOG!%s" % (ANSI_RED, ANSI_RESET))
+            warn("Errors occurred! Read the scrollback!")
```
Comment (ltratt) [inline krun.py]: What happens if errors exceed the scrollback? I thought the point was that we dumped these things to a log file and/or emailed them to a given address too?

Comment (vext01) [inline krun.py]: We've not discussed that, but I don't object. In fact I think it's a good idea.

Let's dump whatever goes via the logging API to a date-stamped file and mail errors as they occur? The latter should be controlled by a config option. Sounds good?

Comment (vext01) [inline krun.py]: If you agree, merge this and I will make a new pull (eventually, I want to spend time on other tasks for a bit now).

Comment (ltratt) [inline krun.py]: Agreed!

**Comment by vext01** [comment]
On Tue, Jul 07, 2015 at 05:19:36AM -0700, Laurence Tratt wrote:
> Merged #10.

Thanks.

-- 
Best Regards
Edd Barrett

http://www.theunixzoo.co.uk

**Comment by vext01** [comment]
RE: standards.

PEP8 is indeed widely adopted. It is mostly pretty good. It has a couple of warts, but not many. We should ask the others what they think.

If you get bored have a play with this:
https://github.com/nvie/vim-flake8

**Pull request merged**